src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,36 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name) {
+    super(health, name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
+      animal.health -= 50;
+    }
+
+    const isAnimalAlive = Animal.alive.filter((item) => item.health > 0);
```

Comment:
Using 'filter' to update 'Animal.alive' is correct, but you should do this after checking and applying the health reduction to ensure that any animal whose health drops to 0 or below is immediately removed from the 'Animal.alive' array.